src/modules/UI/scenes/SendConfirmation/reducer.js

```diff
@@ -1,113 +1,79 @@
 // @flow
 import * as ACTION from './action'
-import * as Constants from '../../../../constants/indexConstants'
-import type { AbcTransaction, AbcParsedUri } from 'airbitz-core-types'
+import type { AbcTransaction } from 'airbitz-core-types'
+import type { AbcMakeSpendInfo } from './selectors'
+import { isEqual } from 'lodash'
 
 export type SendConfirmationState = {
-  transaction: AbcTransaction | null,
-  parsedUri: AbcParsedUri,
-  error: Error | null,
-
-  displayAmount: number,
-  publicAddress: string,
-  feeSatoshi: number,
   label: string,
-
-  // fee: string,
-  feeSetting: string,
-
-  inputCurrencySelected: string,
-  maxSatoshi: number,
-  isPinEnabled: boolean,
-  isSliderLocked: boolean,
-  draftStatus: string,
+  pending: boolean,
   isKeyboardVisible: boolean,
-  pending: boolean
+  transaction: AbcTransaction | null,
+  parsedUri: AbcMakeSpendInfo,
+  error: Error | null
 }
 
 export const initialState: SendConfirmationState = {
-  transaction: null,
-  parsedUri: {
-    publicAddress: '',
-    nativeAmount: ''
+  'label': '',
+  'pending': false,
+  'isKeyboardVisible': false,
+  'transaction': {
+    'txid': '',
+    'date': 0,
+    'currencyCode': '',
+    'blockHeight': -1,
+    'nativeAmount': '',
+    'networkFee': '',
+    'ourReceiveAddresses': [],
+    'signedTx': '',
+    'metadata': {},
+    'otherParams': {}
   },
-  error: null,
-
-  displayAmount: 0,
-  publicAddress: '',
-  feeSatoshi: 0,
-  label: '',
-
-  // fee: '',
-  feeSetting: Constants.STANDARD_FEE,
-
-  inputCurrencySelected: 'fiat',
-  maxSatoshi: 0,
-  isPinEnabled: false,
-  isSliderLocked: false,
-  draftStatus: 'under',
-  isKeyboardVisible: false,
-  pending: false
+  'parsedUri': {
+    'networkFeeOption': 'standard',
+    'customNetworkFee': {},
+    'publicAddress': '',
+    'nativeAmount': '0',
+    'metadata': {
+      'payeeName': '',
+      'category': '',
+      'notes': '',
+      'amountFiat': 0,
+      'bizId': 0,
+      'miscJson': ''
+    }
+  },
+  'error': null
 }
 
 export const sendConfirmation = (state: SendConfirmationState = initialState, action: any) => {
   const { type, data = {} } = action
   switch (type) {
     case ACTION.UPDATE_TRANSACTION: {
-      const transaction: AbcTransaction = data.transaction
-      const parsedUri: AbcParsedUri = data.parsedUri
-      const error: Error = data.error
-      const out: SendConfirmationState = {
+      const { transaction, error } = data
+      return {
         ...state,
         transaction,
-        parsedUri,
         error
       }
-      return out
     }
     case ACTION.UPDATE_PARSED_URI: {
-      const { parsedUri = {} } = data
-      const publicAddress = parsedUri.publicAddress
-      return {
-        ...state,
-        parsedUri,
-        publicAddress
+      const { parsedUri } = data
```

Comment:
This is miss-typed. You actually are using an `AbcMakeSpendInfo` not an `AbcParsedUri` which should really be called a `GuiMakeSpendInfo` since this is not an ABC type but a GUI specific type. Change all references to `makeSpendInfo` including `UPDATE_PARSED_URI` to `MAKE_SPEND_INFO`

Reply:
fixed in https://github.com/Airbitz/edge-react-gui/pull/389